src/unsecure/hash.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::hash::Hasher;
+
+use digest::OutputSizeUser;
+use twox_hash::xxh3::HasherExt;
+
+use crate::hash::{Digest, HashFunction};
+
+/// XXH3 hash function.
+///
+/// Warning: This is NOT a cryptographic hash function and should NOT be used in production.
+#[cfg(feature = "unsecure_schemes")]
+#[derive(Default)]
+pub struct XXH3Unsecure {
+    instance: twox_hash::xxh3::Hash64,
+}
+
+impl OutputSizeUser for XXH3Unsecure {
+    type OutputSize = typenum::U8;
+}
+
+impl HashFunction<typenum::U8> for XXH3Unsecure {
+    fn update(&mut self, data: &[u8]) {
+        self.instance.write(data);
+    }
+
+    fn finalize(self) -> Digest<typenum::U8> {
+        let hash: [u8; 8] = self.instance.finish().to_le_bytes();
+        Digest(hash.into())
+    }
+}
+
+/// XXH128 hash function.
+///
+/// Warning: This is NOT a cryptographic hash function and should NOT be used in production.
+#[cfg(feature = "unsecure_schemes")]
+#[derive(Default)]
+pub struct XXH128Unsecure {
+    instance: twox_hash::xxh3::Hash128,
```

Comment:
Wondering if we should on purpose use a scheme that outputs 32 bytes (so size  is compatible what most chains support): ie SuiAddress and Object derivation in Sui is truncating to 160 bits and having a shorter hash might cause issues.

Reply:
I've considered the same thing. Xxhash currently [does not support 256 bit hashes](https://cyan4973.github.io/xxHash/), and to my knowledge, none of the fast non-cryptographic hash functions does. Bit in this case, we could simply concatenate two 128 bit or pad with zeroes, because we mostly care about avoiding collisions.